go-controller/pkg/ovn/ovn.go

```diff
@@ -348,7 +348,9 @@ func (oc *Controller) Run(wg *sync.WaitGroup, nodeName string) error {
 		}
 		oc.egressFirewallDNS.Run(egressFirewallDNSDefaultDuration)
 		oc.egressFirewallHandler = oc.WatchEgressFirewall()
-
+		if config.Gateway.Mode == config.GatewayModeShared {
```

Comment:
why is this not applicable to local gateway mode?

Reply:
For local gateway mode there is no way to fix this issue (or at least I couldn't think of one). The reason for that is because local gateway mode would require the allow ACL to be placed on the node's switch, since that's where the egress firewall ACLs get programmed. For shared gateway mode it's on the `join` switch, so there's no problem. Given that this allow ACL needs to take precedence over all egress firewall rules (and have a priority of 10001) for connectivity to work to the host network, it would also take precedence over all network policy ACLs (which are also programmed on the node's switch). Hence if a user would create a network policy denying access to the Kubernetes service API, this default allow ACL would overwrite it and grant access (since the egress firewall priority supersede the network policy ones)

Reply:
But what if the there is a risky workload running in a namespace you want to block from accessing the Kube API service? If a NetworkPolicy Egress rule disabled it, would EgressFirewall override and allow it?

Reply:
No, it wouldn't. If the network policy egress rule disables that the packet is dropped from the pipeline and won't be evaluated by any later stage.